### PR TITLE
Conversation AI Chatbot Kit

### PR DESCRIPTION
**Pull request by adrianboguszewski**

### CODE REVIEW

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+<div id="top" align="center">
+  <h1>Conversational AI Chatbot with OpenVINO™ Toolkit</h1>
+  <h4>
+    <a href="https://www.intel.com/content/www/us/en/developer/topic-technology/edge-5g/open-potential.html">🏠&nbsp;About&nbsp;the&nbsp;Kits&nbsp;·</a>
+    <a href="https://www.youtube.com/watch?v=9jnY9PJjYVk">👨‍💻&nbsp;Code&nbsp;Demo&nbsp;Video</a> - Placeholder
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:5]:
```suggestion
    <a href="">👨‍💻&nbsp;Code&nbsp;Demo&nbsp;Video</a> - Placeholder
```

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+To clone the repository, run the following command:
+
+```shell
+git clone https://github.com/openvinotoolkit/openvino_build_deploy.git
+```
+
+The above will clone the repository into a directory named "openvino_build_deploy" in the current directory. Then, navigate into the directory using the following command:
+
+```shell
+cd openvino_build_deploy/ai_ref_kits/conversational-ai-chatbot
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:65]:
```suggestion
cd openvino_build_deploy/ai_ref_kits/conversational_ai_chatbot
```

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+Set Up a Hugging Face Account: If you don't have one, create a [Hugging Face account](https://huggingface.co/welcome).
+
+Authenticate with Meta AI: Go to the LlaMA model (v2 or v3) page on Hugging Face. To authenticate, enter the same email address you used for the Meta AI website. After authentication, you'll gain access to the model.
+
+To use the model, authenticate using the Hugging Face CLI:
+
+```shell
+huggingface-cli login
+```
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:118]: This is no longer needed. They can provide the token when running the script

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+## Model Conversion and Optimization
+
+_NOTE: This reference kit requires much bandwidth and disk space (>8GB) for downloading models. Also, the conversion may take much time (>2h) and need much memory (>32GB)) when running for the first time as the models used here are huge. After the first run, the subsequent runs will be done much faster._
+
+The application uses three separate models for its operation, each requiring conversion and optimization for use with OpenVINO™. Follow the order below to convert and optimize each model:
+
+1. Automated Speech Recognition Distil-Whisper Conversion:
+```shell
+python convert_and_optimize_asr.py --asr_model_type distil-whisper-large-v2 --precision int8
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:131]:
```suggestion
python convert_and_optimize_asr.py --asr_model_type distil-whisper-large-v3 --precision int8
```

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+```
+This script will convert and optimize the automatic speech recognition (ASR) model performing weights quantization.
+
+2. Chat LLama Conversion:
+```shell
+python convert_and_optimize_chat.py --chat_model_type llama3.1-8B --embedding_model_type bge-large --precision int4
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:137]: Add reranker model here

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+```
+This script will handle the conversion and optimization of:
+
+- The chat model (llama3.1-8B) with int4 precision.
+- The embedding model (bge-large) with FP32 precision.
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:142]: Reranker model needed here

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+After running the conversion scripts, you can run app.py to launch the application.
+
+## Running the Application (Gradio Interface)
+
+_NOTE: This application requires much memory (>16GB) as the models used here are huge (especially chatbot). It may also be slow for less powerful devices._
+
+Execute the `app.py` script with the following command, including all necessary model directory arguments:
+```shell
+python app.py --asr_model_dir path/to/asr_model --chat_model_dir path/to/chat_model  --embedding_model path/to/embedding_model --public
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:152]: reranker

File: ai_ref_kits/conversational_ai_chatbot/README.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+### Accessing the Web Interface
+After running the script, Gradio will provide a local URL, typically `http://127.0.0.1:XXXX`, which you can open in your web browser to start interacting with the assistant. If you configured the application to be accessible publicly, Gradio will also provide a public URL.
+
+Trying Out the Application
+1. Navigate to the provided Gradio URL in your web browser.
+2. You will see the Gradio interface with options to input voice.
+3. To interact using voice:
+    - Click on the microphone icon and speak your query.
+    - Wait for the assistant to process your speech and respond.
+4. The assistant will respond to your query in text and audio form.
```
Comment (adrianboguszewski) [inline ai_ref_kits/conversational_ai_chatbot/README.md:165]: Add instructions on file upload and text input

**Review by adrianboguszewski — COMMENTED** [review]
@AnishaUdayakumar

File: ai_ref_kits/conversational_ai_chatbot/README.md (hunk 3 above)
Comment (AnishaUdayakumar) [reply ai_ref_kits/conversational_ai_chatbot/README.md]: updated the requested changes

File: ai_ref_kits/conversational_ai_chatbot/README.md (hunk 5 above)
Comment (AnishaUdayakumar) [reply ai_ref_kits/conversational_ai_chatbot/README.md]: Added reranker

File: ai_ref_kits/conversational_ai_chatbot/README.md (hunk 6 above)
Comment (AnishaUdayakumar) [reply ai_ref_kits/conversational_ai_chatbot/README.md]: updated the requested change.

File: ai_ref_kits/conversational_ai_chatbot/README.md (hunk 7 above)
Comment (AnishaUdayakumar) [reply ai_ref_kits/conversational_ai_chatbot/README.md]: Included the rereanker

File: ai_ref_kits/conversational_ai_chatbot/README.md (hunk 8 above)
Comment (AnishaUdayakumar) [reply ai_ref_kits/conversational_ai_chatbot/README.md]: Updated the instructions based on latest Chat elements.

**Comment by AnishaUdayakumar** [comment]
@adrianboguszewski https://github.com/openvinotoolkit/openvino_build_deploy/blob/conversational-ai-chatbot/ai_ref_kits/conversational_ai_chatbot/README.md

**Comment by adrianboguszewski** [comment]
@AnishaUdayakumar is it still required to get llama license through the Meta website or is it enough to accept license only on Hugging Face?

**Comment by adrianboguszewski** [comment]
https://github.com/user-attachments/assets/262ba878-b845-445b-aff7-0a118517d409

**Comment by raymondlo84** [comment]
closing this? or? @adrianboguszewski

**Pull request merged**